### PR TITLE
add ability to post json bodies to attributes app.

### PR DESCRIPTION
**Pull request by JTCunning**
Currently testless/docless, wanted to see if the POC was up to par.

### CODE REVIEW

File: clustoapi/apps/attribute.py [hunk 1]
```diff
@@ -177,6 +178,26 @@ def add_attr(name):
...
         return util.dumps(msg, status)
 
+    try:
+        json_kwargs = request.json
+    except ValueError as ve:
+        return util.dumps('%s' % (ve,), 400)
+
+    if json_kwargs:
+        # Pull the serialized json tuple off of the params so that we can merge it back in gracefully.
+        for key in kwargs.keys():
```
Comment (JTCunning) [inline clustoapi/apps/attribute.py]: So tldr on this is that request.params give you both forms and querystring, and if you add content-type as json, the querystring turns into a form, so yea, gotta cover both cases and this was then sane-ish way to do it.

File: clustoapi/apps/attribute.py [hunk 2]
```diff
@@ -177,6 +178,26 @@ def add_attr(name):
     if not obj:
         return util.dumps(msg, status)
 
+    try:
+        json_kwargs = request.json
+    except ValueError as ve:
+        return util.dumps('%s' % (ve,), 400)
```
Comment (thekad) [inline clustoapi/apps/attribute.py]: This will always return an error if you *don't* supply a json field, I think?

Comment (JTCunning) [inline clustoapi/apps/attribute.py]: No, request.json will be blank if there is no json-able body. It _will_ return an error if the json is malformed tho.
Current tests pass with this btw, so I'm confident in that statement.

File: clustoapi/apps/attribute.py [hunk 3]
```diff
@@ -177,14 +177,33 @@ def add_attr(name):
...
-        kwargs['number'] = int(kwargs['number'])
-
-    obj.add_attr(**kwargs)
+    try:
+        json_kwargs = request.json
+    except ValueError as ve:
+        return util.dumps('%s' % (ve,), 400)
+
+    if json_kwargs:
+        if request.query:
```
Comment (thekad) [inline clustoapi/apps/attribute.py]: shouldn't you be doing request.forms instead?

**Comment by thekad** [comment]
one question only, lgtm otherwise

File: clustoapi/apps/attribute.py (hunk 3 above)
Comment (JTCunning) [inline clustoapi/apps/attribute.py]: The json body lives in request.forms. Changing to that wouldn't let you post any sort of body.

File: clustoapi/server.py [hunk 4]
```diff
@@ -82,6 +82,7 @@
     'delete': "curl -X DELETE -s -w '\\nHTTP: %{http_code}\\nContent-type: %{content_type}'",
     'delete_i': "curl -X DELETE -si",
     'head': "curl -s -I",
+    'attr_dict': '[{"key":"group","subkey":"admin","value":"apache"},{"key":"group","subkey":"member","value":"webapp"}]'
```
Comment (thekad) [inline clustoapi/server.py]: My only complaint is the name of this variable. Should be something clearly indicated is a sample, sample_json_attrs ?

Comment (JTCunning) [inline clustoapi/server.py]: Totally.

**Pull request merged**